modules/cudacodec/src/ffmpeg_video_source.cpp

```diff
@@ -100,9 +102,34 @@ Codec FourccToCodec(int codec)
     case CV_FOURCC_MACRO('a', 'v', '0', '1'): // fallthru
     case CV_FOURCC_MACRO('A', 'V', '0', '1'): return AV1;
     default:
-        break;
+        return NumCodecs;
     }
+}
+#endif
 
+static
+Codec FourccToCodec(int codec)
+{
+#ifdef _WIN32 // handle old FFmpeg backend - remove when windows shared library is updated
```

Comment:
What FFmpeg version is required? The branch may be relevant for old Ubuntu versions. I propose to add ffmpeg version check instead.